msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AcquireTokenByAuthorizationGrantSupplier.java

```diff
@@ -74,58 +73,52 @@ private boolean IsUiRequiredCacheSupported() {
                 clientApplication instanceof PublicClientApplication;
     }
 
-    private OAuthAuthorizationGrant processPasswordGrant(
-            OAuthAuthorizationGrant authGrant) throws Exception {
-
-        if (!(authGrant.getAuthorizationGrant() instanceof ResourceOwnerPasswordCredentialsGrant)) {
-            return authGrant;
-        }
+    private void processPasswordGrant(OAuthAuthorizationGrant authGrant) throws Exception {
 
-        if (msalRequest.application().authenticationAuthority.authorityType != AuthorityType.AAD) {
-            return authGrant;
+        //Additional processing is only needed if it's a password grant with a non-AAD authority
```

Comment:
User Real discovery needs to be done only for AAD authority. For non-AAD authority, perform the basic grant only.

Reply:
I think you got the condition correct, but the comment wrong?

Reply:
Yep, the if statement correctly exits the method if it's not-AAD, however when writing the comment I just said "non-AAD" because of the `!= AuthorityType.AAD` part

Comment fixed in the latest commit